### PR TITLE
Dra 710 make kaltura xml from solr

### PR DESCRIPTION
**Pull request by thomasegense**
Added manual batch job to create bulk upload XML for Kaltura from solr documents json. There are 3 cases for how to create the ftp download.
bart-tv
bart-radio
kuana/preservica (here extension .mp3 or .mp4 must be added)

### CODE REVIEW

**Review by VictorHarbo — CHANGES_REQUESTED** [review]
Some suggestions

File: src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    }
+
+    // 2  character folders
+    private static String generateBartRadioDownloadUrl(String fileId) {           
+        String pathSplit= fileId.substring(0,2)+"/"+fileId.substring(2,4)+"/"+fileId.substring(4,6)+"/"+fileId;       
+        return FTP_BART_RADIO_PATH+pathSplit;               
+    }
+
+    // 1  character folders + add extension .mp3 or .mp4
+    private static String generateKuanaTVDownloadUrl(String fileId,String resourceType) {
```
[the remark this answers is not in the document]
Comment (thomasegense) [reply src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java:300]: Agree! Fixed

File: src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+
+    //Change for STAGE or PROD! (I do not know PROD values yet)
+    final static int VIDEO_CONVERSION_PROFILE_ID=1406; //This is Kaltura stage for Video
+    final static int AUDIO_CONVERSION_PROFILE_ID=1403; //This is Kaltura stage for Audio
+
+    //The KUANA paths (stage) FTP server has not been established
+    //Name for stage will probably be: deic-download-stage.kb.dk
+    final static String FTP_PRESERVICA_RADIOTV_PATH="https://deic-download.kb.dk/radio-tv/";     
+    final static String FTP_BART_TV_PATH="https://deic-download.kb.dk/kuana-store/bart-access-copies-tv/";
+    final static String FTP_BART_RADIO_PATH="https://deic-download.kb.dk/kuana-store/bart-access-copies-radio/";
```
[the remark this answers is not in the document]
Comment (thomasegense) [reply src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java:75]: Allowed only by DEIC IP

File: src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+    final static int VIDEO_FLAVOR_PARAMS_ID=3;
+    final static int AUDIO_FLAVOR_PARAMS_ID=359;
+
+    //Kaltura enums. Same on stage/prod
+    final static int MEDIATYPE_VIDEO=1;
+    final static int MEDIATYPE_AUDIO=5;
+
+    final static String TAG1="ds-kaltura"; //This should not be changed
+    final static String TAG2="XML-2024-09-23"; //Change this to about current date
+    final static String TAG3="test"; //Asger used this on kaltura-STAGE. Dont know what to use for PROD.
```
[the remark this answers is not in the document]
Comment (VictorHarbo) [reply src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java]: This looks good now

File: src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+
+
+import dk.kb.util.Resolver;
+
+
+/**
+ * The purpose of this class is to parse a Solr document from ds-solr with records and make Kaltura XML's with downloadlink and metadata for bulk upload in Kaltura.
+ * The XML files are then upload to Kaltura manual if there are few, or sent to Petur for very large scale uploads.
+ * 
+ * The solr json document can be produced with this query: holdback_expired_date:[NOW TO *]
```
[the remark this answers is not in the document]
Comment (VictorHarbo) [reply src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java:24]: Please update the solr query here in the written documentation as well.

File: src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (VictorHarbo) [reply src/test/java/dk/kb/datahandler/kaltura/GenerateKalturaUploadXmlFromSolrDocs.java:305]: This still has TV in the method name.

**Review by VictorHarbo — CHANGES_REQUESTED** [review]
Two small things which I believe you might just have overseen.

**Review by VictorHarbo — APPROVED** [review]
Looks good now!

**Pull request merged**